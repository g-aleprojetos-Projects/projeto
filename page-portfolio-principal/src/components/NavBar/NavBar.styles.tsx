import styled from 'styled-components';
import sizes from 'resourses/sizes';
import colors from 'resourses/colors';

export interface PropsRightNav {
  openRightNav?: boolean;
  backgroundDark?: boolean;
  isActive?: boolean;
  idiomaSelecionado?: boolean;
}

export const Container = styled.div`
  display: 'none';

  @media screen and (max-width: ${sizes.media_768}) {
    display: flex;
    position: fixed;
    top: 0;
    width: 100vw;
    height: 100vh;
  }
`;

export const ContainerNav = styled.nav`
  display: flex;
  align-items: center;
  justify-content: center;
  height: ${sizes.px89};
  padding-right: ${sizes.px55};
`;

export const ContainerContent = styled.ul<PropsRightNav>`
  display: flex;
  align-items: center;
  flex-flow: row nowrap;
  position: static;
  background-color: 'transparent';
  transform: 'none';
  top: 'auto';
  right: 'auto';
  width: 'auto';
  height: auto;
  padding-top: 'auto';
  transition: 'none';

  @media screen and (max-width: ${sizes.media_768}) {
    display: flex;
    flex-flow: column nowrap;
    position: fixed;
    background-color: ${colors.black};
    transform: ${props =>
      props.openRightNav ? 'translateX(0)' : 'translateX(100%)'};
    top: 0;
    right: 0;
    width: 100vw;
    padding-top: 3.5rem;
    transition: 'transform 0.3s ease-in-out';
  }
`;

export const Texto = styled.div`
  color: ${colors.white};
`;

export const Menu = styled.div`
  display: none;

  @media screen and (max-width: ${sizes.media_768}) {
    display: flex;
    justify-content: center;
    align-items: center;
    width: 100%;
    padding: 20px 0;
    margin-bottom: 10px;
    background-color: ${colors.tropicalRainForest};
  }
`;

export const Item = styled.li<PropsRightNav>`
  display: flex;
  justify-content: center;
  padding: ${sizes.px8}px;
  margin: ${sizes.px5} ${sizes.px21};
  color: ${colors.white};

  @media screen and (max-width: ${sizes.media_768}) {
    padding: ${sizes.px8}px 0;
  }
`;

export const ItemNav = styled(Item)`
  width: 100%;
  white-space: nowrap;
  cursor: ${props => (props.isActive ? 'default' : 'pointer')};
  :hover {
    background-color: 'none';
  }

  @media screen and (max-width: ${sizes.media_768}) {
    :hover {
      background-color: ${props =>
        props.isActive ? 'none' : colors.blueMagenta};
    }
  }
`;

export const Flag = styled.li`
  display: flex;
  flex-direction: row;
  justify-items: center;
  align-items: center;

  @media screen and (max-width: ${sizes.media_768}) {
    flex-direction: column;
    margin: ${sizes.px5};
  }
`;
