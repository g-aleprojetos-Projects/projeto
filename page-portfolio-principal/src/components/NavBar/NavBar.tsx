import React, {useState} from 'react';
import {useBackgroundContext} from 'context/backgroud';
import {ToggleSwitch} from 'components/toggleSwitch';
import rotas from 'resourses/rotas';
import {useTranslation} from 'react-i18next';
import {namespaces} from '../../utils/i18n';
import '../../utils/i18n';
import {BandeiraBrasil, BandeiraUSA} from 'components/bandeiras';
import * as S from './NavBar.styles';
import {ItemNavBar} from 'components/itemNavBar/ItemNavBar';

type Props = S.PropsRightNav & {
  mudarEstado: () => void;
};

export const NavBar = (props: Props) => {
  const {openRightNav} = props;
  const {i18n} = useTranslation(namespaces.pages.header);
  const {temaEscuro, handleToggle} = useBackgroundContext();
  const [telaAtiva, setTelaAtiva] = useState<string>('Home');

  const handleToggleSwich = () => {
    handleToggle();
  };

  return (
    <S.ContainerNav data-testid={'test_rightNav'}>
      <S.ContainerContent
        data-testid={'test_containerContent'}
        openRightNav={openRightNav}>
        {Object.keys(rotas)
          .slice(0, -1)
          .map((item: string) => {
            return (
              <ItemNavBar
                key={item}
                rota={item}
                telaAtiva={telaAtiva}
                handleTelaAtiva={(item: string): void => {
                  setTelaAtiva(item);
                }}
              />
            );
          })}
        <S.Flag>
          <BandeiraBrasil
            idiomaSelecionado={i18n.resolvedLanguage === 'pt'}
            mudarIdioma={() => i18n.changeLanguage('pt')}
          />
          <BandeiraUSA
            idiomaSelecionado={i18n.resolvedLanguage === 'en'}
            mudarIdioma={() => i18n.changeLanguage('en')}
          />
        </S.Flag>
        <S.Item data-testid={'test_toggle'}>
          <ToggleSwitch isOn={temaEscuro} handleOnclick={handleToggleSwich} />
        </S.Item>
      </S.ContainerContent>
    </S.ContainerNav>
  );
};
