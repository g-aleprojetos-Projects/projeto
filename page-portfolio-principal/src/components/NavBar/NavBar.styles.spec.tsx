import React from 'react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';
import 'jest-styled-components';
import * as S from './NavBar.styles';

describe('RightNav.styles', () => {
  test('ContainerNav DEVE ser igual ao snapshot', () => {
    const containerNav = renderer.create(<S.ContainerNav />).toJSON();
    expect(containerNav).toMatchSnapshot();
  });
  test('ContainerContent DEVE ser igual ao snapshot QUANDO "openRightNav" for false e "backgroundDark" for false.', () => {
    const containerContent = renderer
      .create(
        <S.ContainerContent openRightNav={false} backgroundDark={false} />,
      )
      .toJSON();
    expect(containerContent).toMatchSnapshot();
  });
  test('ContainerContent DEVE ser igual ao snapshot QUANDO "openRightNav" for false e "backgroundDark" for true.', () => {
    const containerContent = renderer
      .create(<S.ContainerContent openRightNav={false} backgroundDark={true} />)
      .toJSON();
    expect(containerContent).toMatchSnapshot();
  });
  test('ContainerContent DEVE ser igual ao snapshot QUANDO "openRightNav" for true e "backgroundDark" for false.', () => {
    const containerContent = renderer
      .create(<S.ContainerContent openRightNav={true} backgroundDark={false} />)
      .toJSON();
    expect(containerContent).toMatchSnapshot();
  });
  test('ContainerContent DEVE ser igual ao snapshot QUANDO "openRightNav" for true e "backgroundDark" for true.', () => {
    const containerContent = renderer
      .create(<S.ContainerContent openRightNav={true} backgroundDark={true} />)
      .toJSON();
    expect(containerContent).toMatchSnapshot();
  });

  test('Texto DEVE ser igual ao snapshot', () => {
    const texto = renderer.create(<S.Texto />).toJSON();
    expect(texto).toMatchSnapshot();
  });
  test('Menu DEVE ser igual ao snapshot ', () => {
    const menu = renderer.create(<S.Menu />).toJSON();
    expect(menu).toMatchSnapshot();
  });

  test('Item DEVE ser igual ao snapshot.', () => {
    const item = renderer.create(<S.Item />).toJSON();
    expect(item).toMatchSnapshot();
  });
  test('ItemNav DEVE ser igual ao snapshot QUANDO "isActive" for false.', () => {
    const item = renderer.create(<S.Item isActive={false} />).toJSON();
    expect(item).toMatchSnapshot();
  });
  test('ItemNav DEVE ser igual ao snapshot QUANDO "isActive" for true.', () => {
    const item = renderer.create(<S.Item isActive={true} />).toJSON();
    expect(item).toMatchSnapshot();
  });
  test('Flag DEVE ser igual ao snapshot.', () => {
    const flag = renderer.create(<S.Flag />).toJSON();
    expect(flag).toMatchSnapshot();
  });
});
