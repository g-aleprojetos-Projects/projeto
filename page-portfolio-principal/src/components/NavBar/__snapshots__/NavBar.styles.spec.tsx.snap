// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RightNav.styles ContainerContent DEVE ser igual ao snapshot QUANDO "openRightNav" for false e "backgroundDark" for false. 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-flow: row nowrap;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
  position: static;
  background-color: 'transparent';
  -webkit-transform: 'none';
  -ms-transform: 'none';
  transform: 'none';
  top: 'auto';
  right: 'auto';
  width: 'auto';
  height: auto;
  padding-top: 'auto';
  -webkit-transition: 'none';
  transition: 'none';
}

@media screen and (max-width:768px) {
  .c0 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-flow: column nowrap;
    -ms-flex-flow: column nowrap;
    flex-flow: column nowrap;
    position: fixed;
    background-color: #000000;
    -webkit-transform: translateX(100%);
    -ms-transform: translateX(100%);
    transform: translateX(100%);
    top: 0;
    right: 0;
    width: 100vw;
    padding-top: 3.5rem;
    -webkit-transition: 'transform 0.3s ease-in-out';
    -webkit-transition: 'transform 0.3s ease-in-out';
    transition: 'transform 0.3s ease-in-out';
  }
}

<ul
  className="c0"
/>
`;

exports[`RightNav.styles ContainerContent DEVE ser igual ao snapshot QUANDO "openRightNav" for false e "backgroundDark" for true. 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-flow: row nowrap;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
  position: static;
  background-color: 'transparent';
  -webkit-transform: 'none';
  -ms-transform: 'none';
  transform: 'none';
  top: 'auto';
  right: 'auto';
  width: 'auto';
  height: auto;
  padding-top: 'auto';
  -webkit-transition: 'none';
  transition: 'none';
}

@media screen and (max-width:768px) {
  .c0 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-flow: column nowrap;
    -ms-flex-flow: column nowrap;
    flex-flow: column nowrap;
    position: fixed;
    background-color: #000000;
    -webkit-transform: translateX(100%);
    -ms-transform: translateX(100%);
    transform: translateX(100%);
    top: 0;
    right: 0;
    width: 100vw;
    padding-top: 3.5rem;
    -webkit-transition: 'transform 0.3s ease-in-out';
    -webkit-transition: 'transform 0.3s ease-in-out';
    transition: 'transform 0.3s ease-in-out';
  }
}

<ul
  className="c0"
/>
`;

exports[`RightNav.styles ContainerContent DEVE ser igual ao snapshot QUANDO "openRightNav" for true e "backgroundDark" for false. 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-flow: row nowrap;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
  position: static;
  background-color: 'transparent';
  -webkit-transform: 'none';
  -ms-transform: 'none';
  transform: 'none';
  top: 'auto';
  right: 'auto';
  width: 'auto';
  height: auto;
  padding-top: 'auto';
  -webkit-transition: 'none';
  transition: 'none';
}

@media screen and (max-width:768px) {
  .c0 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-flow: column nowrap;
    -ms-flex-flow: column nowrap;
    flex-flow: column nowrap;
    position: fixed;
    background-color: #000000;
    -webkit-transform: translateX(0);
    -ms-transform: translateX(0);
    transform: translateX(0);
    top: 0;
    right: 0;
    width: 100vw;
    padding-top: 3.5rem;
    -webkit-transition: 'transform 0.3s ease-in-out';
    -webkit-transition: 'transform 0.3s ease-in-out';
    transition: 'transform 0.3s ease-in-out';
  }
}

<ul
  className="c0"
/>
`;

exports[`RightNav.styles ContainerContent DEVE ser igual ao snapshot QUANDO "openRightNav" for true e "backgroundDark" for true. 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-flow: row nowrap;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
  position: static;
  background-color: 'transparent';
  -webkit-transform: 'none';
  -ms-transform: 'none';
  transform: 'none';
  top: 'auto';
  right: 'auto';
  width: 'auto';
  height: auto;
  padding-top: 'auto';
  -webkit-transition: 'none';
  transition: 'none';
}

@media screen and (max-width:768px) {
  .c0 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-flow: column nowrap;
    -ms-flex-flow: column nowrap;
    flex-flow: column nowrap;
    position: fixed;
    background-color: #000000;
    -webkit-transform: translateX(0);
    -ms-transform: translateX(0);
    transform: translateX(0);
    top: 0;
    right: 0;
    width: 100vw;
    padding-top: 3.5rem;
    -webkit-transition: 'transform 0.3s ease-in-out';
    -webkit-transition: 'transform 0.3s ease-in-out';
    transition: 'transform 0.3s ease-in-out';
  }
}

<ul
  className="c0"
/>
`;

exports[`RightNav.styles ContainerNav DEVE ser igual ao snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 5.563rem;
  padding-right: 3.438rem;
}

<nav
  className="c0"
/>
`;

exports[`RightNav.styles Flag DEVE ser igual ao snapshot. 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  justify-items: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

@media screen and (max-width:768px) {
  .c0 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    margin: 0.313rem;
  }
}

<div
  className="c0"
/>
`;

exports[`RightNav.styles Item DEVE ser igual ao snapshot. 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0.5rempx;
  margin: 0.313rem 1.313rem;
  color: #ffffff;
}

@media screen and (max-width:768px) {
  .c0 {
    padding: 0.5rempx 0;
  }
}

<li
  className="c0"
/>
`;

exports[`RightNav.styles ItemNav DEVE ser igual ao snapshot QUANDO "isActive" for false. 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0.5rempx;
  margin: 0.313rem 1.313rem;
  color: #ffffff;
}

@media screen and (max-width:768px) {
  .c0 {
    padding: 0.5rempx 0;
  }
}

<li
  className="c0"
/>
`;

exports[`RightNav.styles ItemNav DEVE ser igual ao snapshot QUANDO "isActive" for true. 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0.5rempx;
  margin: 0.313rem 1.313rem;
  color: #ffffff;
}

@media screen and (max-width:768px) {
  .c0 {
    padding: 0.5rempx 0;
  }
}

<li
  className="c0"
/>
`;

exports[`RightNav.styles Menu DEVE ser igual ao snapshot  1`] = `
.c0 {
  display: none;
}

@media screen and (max-width:768px) {
  .c0 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    width: 100%;
    padding: 20px 0;
    margin-bottom: 10px;
    background-color: #00875F;
  }
}

<div
  className="c0"
/>
`;

exports[`RightNav.styles Texto DEVE ser igual ao snapshot 1`] = `
.c0 {
  color: #ffffff;
}

<div
  className="c0"
/>
`;
